src/Hcal/HcalDigiProducer.cxx

```diff
@@ -1,392 +1,235 @@
-
-      // Note the side Hcal doesn't have super strips
-      // This is the quantized position along the length of the bar - LEFT/RIGHT
-      // is for fixed HCAL geometry
-      // float ecal_width_(525);
-      // if (cur_subsection == ldmx::HcalID::TOP)    cur_xpos =
-      // half_total_width/2.0
-      // - ecal_width/4.0; if (cur_subsection == ldmx::HcalID::BOTTOM) cur_xpos
-      // = -half_total_width/2.0 + ecal_width/4.0; if (cur_subsection ==
-      // ldmx::HcalID::LEFT)   cur_ypos =  half_total_width/2.0 -
-      // ecal_width/4.0; if (cur_subsection == ldmx::HcalID::RIGHT)  cur_ypos =
-      // -half_total_width/2.0 + ecal_width/4.0;
-
-      // This would be the quantized z position. The side_hcal_z0 value must be
-      // derived fromn the geometry! float side_hcal_z0(215.5); cur_zpos =
-      // side_hcal_z0+(cur_strip+0.5)*strip_width ;
-
-      // This is the quantized position along the thickness of the bar - check
-      // RIGHT / LEFT float back_hcal_layer_thickness(39); float
-      // side_hcal_xy_offset(294); if (cur_subsection == ldmx::HcalID::TOP)
-      // cur_ypos = side_hcal_xy_offset+(cur_layer-1)*back_hcal_layer_thickness;
-      // if (cur_subsection == ldmx::HcalID::BOTTOM) cur_ypos =
-      // -side_hcal_xy_offset-(cur_layer-1)*back_hcal_layer_thickness; if
-      // (cur_subsection == ldmx::HcalID::LEFT)   cur_xpos =
-      // -side_hcal_xy_offset-(cur_layer-1)*back_hcal_layer_thickness; if
-      // (cur_subsection == ldmx::HcalID::RIGHT)  cur_xpos =
-      // side_hcal_xy_offset+(cur_layer-1)*back_hcal_layer_thickness;
     }
+  }
 
-    if (hcalLayerPEs[detIDraw] >= readoutThreshold_) {
-      ldmx::HcalHit hit;
-      hit.setID(detIDraw);
-      hit.setPE(hcalLayerPEs[detIDraw]);
-      hit.setMinPE(hcalLayerMinPEs[detIDraw]);
-      hit.setAmplitude(hcalLayerPEs[detIDraw]);
-      hit.setEnergy(energy);
-      hit.setTime(hcaldetIDTime[detIDraw]);
-      hit.setXPos(cur_xpos);  // quantized and smeared positions
-      hit.setYPos(cur_ypos);  // quantized and smeared positions
-      hit.setZPos(cur_zpos);
-      hit.setNoise(false);
-
-      hcalRecHits.push_back(hit);
+  /******************************************************************************************
+   * Noise Simulation on Empty Channels
+   *****************************************************************************************/
+  if (noise_) {
+    int numChannels = 0;
+    for (int l = 0; l < hcalGeometry.getNumSections(); l++) {
+      int nChannels =
+          hcalGeometry.getNumLayers(l) * hcalGeometry.getNumStrips(l);
+      if (l == 0) nChannels *= 2;
+      numChannels += nChannels;
     }
+    int numEmptyChannels = numChannels - hcalDigis.getNumDigis();
+    // noise generator gives us a list of noise amplitudes [mV] that randomly
+    // populate the empty channels and are above the readout threshold
+    auto noiseHitAmplitudes{
+        noiseGenerator_->generateNoiseHits(numEmptyChannels)};
```

Comment:
This noise generator has not been configured. This is dangerous because the defaults for the noise generation may or may not match your intended purpose.

https://github.com/LDMX-Software/ldmx-sw/blob/925d12ccaadc5d76a6025247aae8f173543db47c/Tools/include/Tools/NoiseGenerator.h#L68-L75

You should define the threshold, mean noise, and pedestal in the units you expect. For example, you set the voltage equal to the noise amplitude plus some other voltages. This means you are assuming the noise generator is generating amplitudes in units of mV. Configuring the noise generator should happen in your `configure` method.

Reply:
Good catch, now implemented [here](https://github.com/LDMX-Software/Hcal/blob/iss1/src/Hcal/HcalDigiProducer.cxx#L56-L62):
```
 noiseGenerator_->setNoise(
      hgcrocParams.getParameter<double>("noiseRMS"));  // rms noise in mV
  noiseGenerator_->setPedestal(
      gain_ * pedestal_);  // mean noise amplitude (if using Gaussian Model for
                           // the noise) in mV
  noiseGenerator_->setNoiseThreshold(
      gain_ * readoutThreshold_);  // threshold for readout in mV
```.
I have also tested that the hcalDigiProducer works with noise by uncommenting 
```hcalDigis.hgcroc.noise = False```
in the test config.